### DISCUSSION TITLE
Feature Request: Table and column comments

### DISCUSSION THREAD

**Discussion by tekumara** [duckdb/duckdb#6290] [Ideas]
Support table and column comments similar to postgres/snowflakedb.

This allows storing arbitrary metadata about table and columns. It could be useful for governance reasons to ask questions like, does the column contain personally identifiable information (PII)? who is the owner of this table?

Thank you!

**Comment by shippy** [comment]
Seconded. `persist_docs` is my favorite dbt feature; I was sad to find that `dbt-duckdb` didn't support it because DuckDB doesn't.

**Comment by adriens** [comment]
:heavy_plus_sign: :one: As it would help better use LLM on to pof data : describe columns and tables in a way we can make LLM more efficient to create smarter queries :pray:

**Comment by l1t1** [comment]
maybe you can save them in a table as a workaround

**Reply by adriens** [reply]
Indeed, but sotring them as metadata open many other benefits... like getting them from jdbc drivers as part of schema metadata :smile_cat:

**Comment by JimCallahanOrlando** [comment]
COMMENT command & implementation with pg_description system table.
In Postgres the COMMENT feature is implemented with a Postgres System Catalog (not the Information Schema).
"The [Postgres system] catalog pg_description stores optional descriptions (comments) for each database object. Descriptions can be manipulated with the [COMMENT](https://www.postgresql.org/docs/current/sql-comment.html) command and viewed with psql's \d commands. Descriptions of many built-in system objects are provided in the initial contents of pg_description."
https://www.postgresql.org/docs/current/catalog-pg-description.html

**Comment by JimCallahanOrlando** [comment]
COMMENTS
Comments are useful for:
a. Description 
b. Units
c. Source of data (UN, U.S. Census, U.S. CIA, UCI, Kaggle)
d. Confidentiality (GDPR, HIPPA, Personally Identifiable Information, Top Secret, Trade Secret, Corporate Confidential or copyright)
e. Vintage (when was the data collected or what is its reference date "as of")
https://www.dataquest.io/blog/free-datasets-for-projects/

**Reply by adriens** [reply]
... and to automate custom smart PROMPT generation for RAG (for agents)

**Comment by JimCallahanOrlando** [comment]
Stack Overflow: How to retrieve comments in a PostgreSQL database?
https://stackoverflow.com/questions/11493978/how-to-retrieve-the-comment-of-a-postgresql-database

**Comment by JimCallahanOrlando** [comment]
YouTube: "Comments in Postgres" -- includes demo in psql and some use cases
https://www.youtube.com/watch?v=EUUo8AzQAnI

**Comment by adriens** [comment]
btw
- https://github.com/duckdb/duckdb/issues/6759

**Comment by carlopi** [comment]
Since https://github.com/duckdb/duckdb/pull/10372 landed, COMMENT should be available on nightly version of duckdb, if you want to give it a try.

**Reply by adriens** [reply]
You mean... it's available on [`v0.9.2`](https://github.com/duckdb/duckdb/releases/tag/v0.9.2)  :grey_question:

**Reply by Tishj** [reply]
No, nightly is prerelease

**Reply by carlopi** [reply]
You should be able to install nightly packages and try out COMMENT ON functionality, see https://duckdb.org/docs/archive/0.9.2/installation/?version=main&environment=cli&installer=binary

**Comment by Mytherin** [comment]
This has now been implemented

**Reply by adriens** [reply]
Can't wait to see it released and make my PROMPT generation even better and automatic :sloth: :star_struck:
